### PR TITLE
feat(aom): add new resource alarm silence rule

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/aom' TESTARGS='-run TestAccAlarmSilenceRule_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/aom -v -run TestAccAlarmSilenceRule_basic -timeout 360m -parallel 4
=== RUN   TestAccAlarmSilenceRule_basic
=== PAUSE TestAccAlarmSilenceRule_basic
=== CONT  TestAccAlarmSilenceRule_basic
--- PASS: TestAccAlarmSilenceRule_basic (21.46s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/aom       21.560s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_alarm_silence_rule_test.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+		},
+	})
+}
+
+func testAlarmSilenceRule_basic(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_aom_alarm_silence_rule" "test" {
+  name = "%s"
+  description = "terraform test"
+  time_zone    = "Asia/Shanghai"
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_alarm_silence_rule_test.go]: align equal signs

File: huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_alarm_silence_rule_test.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+}
+`, name)
+}
+
+func testAlarmSilenceRule_basic_update(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_aom_alarm_silence_rule" "test" {
+  name = "%s"
+  description = "terraform test update"
+  time_zone    = "Asia/Shanghai"
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_alarm_silence_rule_test.go]: ditto

File: huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_alarm_silence_rule_test.go (hunk 1 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_alarm_silence_rule_test.go]: fixed

File: huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_alarm_silence_rule_test.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_alarm_silence_rule_test.go]: fixed

File: docs/resources/aom_alarm_silence_rule.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+* `silence_time` - (Required, List) Specifies the silence time of the rule.
+  The [silence_time](#silence_time) structure is documented below.
+
+* `silence_conditions` - (Required, List) Specifies the silence conditions of the rule.
+  Different silence conditions are parallel. A maximum of 10 silence conditions are allowed.
+  The [silence_conditions](#silence_conditions) structure is documented below.
+
+* `description` - (Optional, String) Specifies the description.
+
+* `user_id` - (Optional, String, ForceNew) Specifies the user ID.
```
Comment (ShiChangkuo) [inline docs/resources/aom_alarm_silence_rule.md]: user_id is not defined in schema

File: huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_alarm_silence_rule_test.go [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+					resource.TestCheckResourceAttr(rName, "silence_conditions.0.conditions.0.key", "event_severity"),
+					resource.TestCheckResourceAttr(rName, "silence_conditions.0.conditions.0.operate", "EQUALS"),
+				),
+			},
+			{
+				Config: testAlarmSilenceRule_basic_update(name),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(rName, "name", name),
+					resource.TestCheckResourceAttr(rName, "name", name),
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_alarm_silence_rule_test.go]: repeat test check

File: huaweicloud/services/aom/resource_huaweicloud_aom_alarm_silence_rule.go [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+				Elem:     AlarmSilenceRuleSilenceTimeSchema(),
+				Required: true,
+			},
+			"silence_conditions": {
+				Type:        schema.TypeList,
+				Elem:        AlarmSilenceRuleSilenceConditionsSchema(),
+				Required:    true,
+				Description: `Specifies the silence conditions of the rule.`,
+			},
+			"description": {
```
Comment (deer-hang) [inline huaweicloud/services/aom/resource_huaweicloud_aom_alarm_silence_rule.go:67]: the description in console can update to empty. Please check if the field need to mark as computed

File: docs/resources/aom_alarm_silence_rule.md (hunk 3 above)
Comment (Jason-Zhang9309) [reply docs/resources/aom_alarm_silence_rule.md]: fixed

File: huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_alarm_silence_rule_test.go (hunk 4 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_alarm_silence_rule_test.go]: fixed

File: huaweicloud/services/aom/resource_huaweicloud_aom_alarm_silence_rule.go (hunk 5 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/aom/resource_huaweicloud_aom_alarm_silence_rule.go:67]: as discussed, keep it as it is

**Pull request merged**